### ISSUE TITLE
Suggestion is cut off when MentionsInput is close to the bottom of the window

### ISSUE THREAD

**Issue by coxom** [signavio/react-mentions#269]
**Steps to reproduce:**

1. Position an MentionsInput somewhere close to the bottom edge of the window
2. Open the mention list for the input

**Expected behaviour:**
![image](https://user-images.githubusercontent.com/35224636/45036346-24721d80-b054-11e8-93d2-d40bc06d92a7.png)

**Observed behaviour:**

![image](https://user-images.githubusercontent.com/35224636/45036310-099fa900-b054-11e8-96c7-33a5294c6015.png)

**Workaround:**
We added this snippet to the code line 647:

```javascript
// guard for mentions suggestions list clipped by bottom edge of window
const tooltipBottom = 
   _this3.containerRef.getBoundingClientRect().top +
   caretPosition.top +
   suggestions.offsetHeight;
const viewportHeight = Math.max(document.documentElement.clientHeight, window.innerHeight || 0);

if (tooltipBottom > viewportHeight) {
    position.bottom = _this3.containerRef.offsetHeight - caretPosition.top;
} else {
    position.top = caretPosition.top - highlighter.scrollTop;
}
```

Is react-mentions supposed to support this use case? Should we make a pull request?

**Comment by coxom** [comment]
@frontendphil as soon as I can I will make a PR. I used the same code style used for guard right edge of window.
It's setting the top value in the state that gets used by SuggestionsOverlay (this.state.suggestionsPosition .

**Comment by DJTB** [comment]
I'd love to see this - coming up against it right now (textarea is located at bottom right of viewport).

@coxom Should I write my own PR - or wait for you if you won't be long?
No pressure - just don't want us to _both_ spend time on it.

**Comment by coxom** [comment]
@DJTB Sorry I got stuck in the tests and then I had to move to other tasks. I opened the pull request with the fix I tested: https://github.com/signavio/react-mentions/pull/273

**Comment by kashifshamaz21** [comment]
Seems like the PR in #339 should fix this issue. For others who stumble upon this issue, the fix done in #339 should resolve it. So lets not raise more tickets on this issue, comment on the mentioned PR if needed.
